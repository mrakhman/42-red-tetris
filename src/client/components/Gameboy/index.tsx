import { StyleSheet, Text, View } from 'react-native';
import * as React from 'react';
import Keypad from '/client/components/Keypad';

import { useWindowDimensions } from '/client/hooks/useWindowDimensions';

export default function Gameboy({ children }: { children: JSX.Element[] | JSX.Element }): JSX.Element {
  const window = useWindowDimensions();
  const w = window.width;
  const h = window.height;
  const ratio = h / w;
  let scale;
  if (ratio < 1.5) {
    scale = h / 960;
  } else {
    scale = w / 560;
  }
  return (
    <View style={styles.container}>
      <View style={[styles.gameboy, { transform: [{ scale }] }]}>
        <Text style={styles.title}>R E D â–  T E T R I S</Text>
        <View style={styles.display}>
          {children}
        </View>
        <Keypad/>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#009688',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 20,
    height: '100%',
  },
  display: {
    alignItems: 'center',
    backgroundColor: '#9ead86',
    borderBottomColor: '#fa6b6b',
    borderLeftColor: '#980f0f',
    borderRightColor: '#fa6b6b',
    borderTopColor: '#980f0f',
    borderWidth: 5,
    height: 470,
    justifyContent: 'center',
    padding: 10,
    width: 400,
  },
  gameboy: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#af0000',
    borderRadius: 20,
    paddingBottom: 50,
    paddingHorizontal: 50,
  },
  title: {
    fontSize: 30,
    marginVertical: 20,
    textAlign: 'center',
  },
});
