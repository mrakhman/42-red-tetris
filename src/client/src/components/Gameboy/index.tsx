import { StyleSheet, Text, View } from 'react-native';
import * as React from 'react';
import Keypad from '/components/Keypad';

export default function Gameboy({ children }: { children: JSX.Element[] | JSX.Element }): JSX.Element {
  return (
    <View style={styles.container}>
      <View style={styles.gameboy}>
        <Text style={styles.title}>R E D â–  T E T R I S</Text>
        <View style={styles.display}>
          {children}
        </View>
        <Keypad/>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#009688',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 20,
  },
  display: {
    alignItems: 'center',
    backgroundColor: '#9ead86',
    borderBottomColor: '#fa6b6b',
    borderLeftColor: '#980f0f',
    borderRightColor: '#fa6b6b',
    borderTopColor: '#980f0f',
    borderWidth: 5,
    height: 470,
    justifyContent: 'center',
    padding: 10,
    width: 400,
  },
  gameboy: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#af0000',
    borderRadius: 20,
    paddingBottom: 50,
    paddingHorizontal: 50,
  },
  title: {
    fontSize: 30,
    marginVertical: 20,
    textAlign: 'center',
  },
});
