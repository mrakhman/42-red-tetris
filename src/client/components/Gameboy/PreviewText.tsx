import React from 'react';
import { Text, View } from 'react-native';
import { styles } from '/client/components/Gameboy';

type Props = {
  isMultiplayerMode: boolean,
  opponentsNumber: number,
  isLeader?: boolean,
  gameover?: boolean,
}

const PreviewText = (props: Props): JSX.Element  => {
  const { isMultiplayerMode, opponentsNumber, isLeader, gameover } = props;
  const secondLineText = (opponentsNumber: number, isLeader?: boolean) => {
    if (opponentsNumber <= 0) return 'Wait for other players';
    if (opponentsNumber > 0 && isLeader) return 'Press Start to begin';
    if (opponentsNumber > 0 && !isLeader) return 'Wait for leader to start the game';
  };

  if (gameover) {
    return (
      <View style={{ alignItems: 'center' }}>
        <Text style={styles.gameover}>Game Over</Text>
        <Text style={styles.gameMode}>Please wait for other players to finish</Text>
      </View>
    );
  }
  if (isMultiplayerMode) {
    return (
      <View style={{ alignItems: 'center' }}>
        <Text style={styles.gameMode}>You are in multiplayer mode</Text>
        <Text style={styles.gameMode}>{opponentsNumber} other player(s) in your room</Text>
        <Text>{secondLineText(opponentsNumber, isLeader)}</Text>
      </View>
    );
  }
  return (
    <View style={{ alignItems: 'center' }}>
      <Text style={styles.gameMode}>You are in solo mode</Text>
      <Text>Press Play(P) to start</Text>
    </View>
  );
};

export default PreviewText;
