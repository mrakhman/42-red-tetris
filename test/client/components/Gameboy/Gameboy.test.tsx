import React from 'react';
import { Text, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Gameboy from '/client/components/Gameboy';

describe('Snapshots', () => {
  it('should render Gameboy', () => {
    const wrapper = shallow(<Gameboy><View><Text>aaa</Text></View></Gameboy>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
