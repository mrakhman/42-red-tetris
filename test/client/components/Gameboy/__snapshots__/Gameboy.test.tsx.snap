// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render Gameboy 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1f393e",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#990011",
          "borderRadius": 20,
          "justifyContent": "center",
          "paddingBottom": 30,
          "paddingHorizontal": 30,
        },
        Object {
          "transform": Array [
            Object {
              "scale": 1.3392857142857142,
            },
          ],
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
          "marginVertical": 15,
          "textAlign": "center",
        }
      }
    >
      R E D â–  T E T R I S
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#9ead86",
          "borderBottomColor": "#fa6b6b",
          "borderLeftColor": "#980f0f",
          "borderRightColor": "#fa6b6b",
          "borderTopColor": "#980f0f",
          "borderWidth": 5,
          "height": 500,
          "justifyContent": "center",
          "padding": 10,
          "width": 440,
        }
      }
    >
      <View>
        <Text>
          aaa
        </Text>
      </View>
    </View>
    <Keypad
      opponentsNumber={0}
    />
  </View>
</View>
`;
